Raven.Client.Lightweight/Connection/ReplicationInformer.cs

```diff
@@ -72,20 +79,22 @@ private Task UpdateReplicationInformationIfNeededInternalAsync(string url, Func<
                 firstTime = false;
 
                 if (lastReplicationUpdate.Add(Conventions.TimeToWaitBetweenReplicationTopologyUpdates) > SystemTime.UtcNow && force == false)
-                    return new CompletedTask();
+                    return;
 
-                var taskCopy = refreshReplicationInformationTask;
-                if (taskCopy != null)
-                    return taskCopy;
+                if (refreshReplicationInformationTask != null)
+                    await refreshReplicationInformationTask.ConfigureAwait(false);
```

Comment:
If I understood correctly next refresh should await the previous refresh task if available and then return to not trigger subsequent refreshes